### ISSUE TITLE
BrowserHttpHandler can't stream large responses even with SetBrowserResponseStreamingEnabled

### ISSUE THREAD

**Issue by SteveSandersonMS** [dotnet/runtime#60287]
### Description

On WebAssembly, `HttpClient.SendAsync` can't handle large responses (e.g., > 1GB) and fails with errors like

```
Uncaught (in promise) ExitStatus {name: 'ExitStatus', message: 'Program terminated with exit(0)', status: 0}
```

while also logging an absolutely *giant* stack trace (*many* screens high, dozens or maybe hundreds) starting with stuff like this:

```
580530 | @ | dotnet.6.0.0-rtm.21508.5.38wx3tlowm.js:1
  | _emscripten_asm_const_int | @ | dotnet.6.0.0-rtm.21508.5.38wx3tlowm.js:1
  | $func5907 | @ | 00945abe:0x14ccd8
  | $func8601 | @ | 00945abe:0x1b46be
  | $func3936 | @ | 00945abe:0xd198a
  | $func1127 | @ | 00945abe:0x29bd1
  | $func106 | @ | 00945abe:0x515c
  | $func2788 | @ | 00945abe:0x8fb78
  | $func119 | @ | 00945abe:0x56b4
  | $func159 | @ | 00945abe:0x6120
  | $func878 | @ | 00945abe:0x1dc5f
  | $func181 | @ | 00945abe:0x6403
  | $func2927 | @ | 00945abe:0x94e01
  | $func761 | @ | 00945abe:0x1952b
  | $func163 | @ | 00945abe:0x6181
  | $func2307 | @ | 00945abe:0x6fdca
  | $func7015 | @ | 00945abe:0x17bac4
  | $func6830 | @ | 00945abe:0x172bc5
  | $func3524 | @ | 00945abe:0xb47cc
  | $func675 | @ | 00945abe:0x16602
  | $fb | @ | 00945abe:0x16f60e
  | Module._mono_wasm_invoke_method | @ | dotnet.6.0.0-rtm.21508.5.38wx3tlowm.js:1
  | managed_BINDINGS_SetTaskSourceResult | @ | managed_BINDINGS_SetTaskSourceResult:17
  | (anonymous) | @ | dotnet.6.0.0-rtm.21508.5.38wx3tlowm.js:1
  | Promise.then (async) |   |  
  | _wrap_js_thenable_as_task | @ | dotnet.6.0.0-rtm.21508.5.38wx3tlowm.js:1
  | _js_to_mono_obj | @ | dotnet.6.0.0-rtm.21508.5.38wx3tlowm.js:1
  | _mono_wasm_invoke_js_with_args | @ | dotnet.6.0.0-rtm.21508.5.38wx3tlowm.js:1
  | $func2588 | @ | 00945abe:0x80614
  | $func2307 | @ | 00945abe:0x689a7
  | $func7015 | @ | 00945abe:0x17bac4
  | $func6830 | @ | 00945abe:0x172bc5
  | $func3524 | @ | 00945abe:0xb47cc
  | $func675 | @ | 00945abe:0x16602
  | $fb | @ | 00945abe:0x16f60e
  | Module._mono_wasm_invoke_method | @ | dotnet.6.0.0-rtm.21508.5.38wx3tlowm.js:1
  | managed_BINDINGS_SetTaskSourceResult | @ | managed_BINDINGS_SetTaskSourceResult:17
  | (anonymous) | @ | dotnet.6.0.0-rtm.21508.5.38wx3tlowm.js:1
  | Promise.then (async) | 
```

This was originally reported at https://github.com/dotnet/aspnetcore/issues/36872

### Reproduction Steps

Using code like the following on WebAssembly:

```cs
var req = new HttpRequestMessage(HttpMethod.Get, someUrl));	
req.SetBrowserResponseStreamingEnabled(true);
var response = await HttpClient.SendAsync(req); // Crashes here with big files
Console.WriteLine(response.Content.Headers.ContentLength);
```

... it's possible to get a `Stream` representing the body of an HTTP response, as long as it's pretty small. For larger responses (tested with 1 GB), it fails with the above error.

Note that I also tried without `SetBrowserResponseStreamingEnabled`, and it also crashed, but with the same error (`Uncaught (in promise) ExitStatus {name: 'ExitStatus', message: 'Program terminated with exit(0)', status: 0}` but without the giant stack trace.

### Expected behavior

Should be possible to read from arbitrary-sized response streams, assuming the developer doesn't choose to read it all into memory at once.

### Actual behavior

Crashes as described above

### Regression?

Don't know

### Known Workarounds

None

### Configuration

Blazor WebAssembly on SDK 6.0.100-rtm.21508.19, any browser

### Other information

Based on real customer issue: https://github.com/dotnet/aspnetcore/issues/36872

**Comment by pavelsavara** [comment]
^^ FYI @ylr-research

**Comment by ylr-research** [comment]
@pavelsavara thanks so much for the suggestion. @SteveSandersonMS thanks for following up on this issue. I was able to successfully download the file with both `JavaScript` and `HttpClient`. 
@SteveSandersonMS for JS function I used the one you recommended -see below- and the response time for 1GB was ~11s.
```
        async function download(url, fileName) {
            var link = document.createElement('a');
            link.setAttribute('href', url);
            link.setAttribute('download', fileName);
            link.click();
        }
```
On the other hand, same file using `HttpClient` was ~39s. Although the JS function is 2x times faster I see `HttpClient` has shown great improvements. See below the snapshot for reference.

<img width="497" alt="dotnetstream" src="https://user-images.githubusercontent.com/63487874/140596186-440268b7-af11-48c0-9259-9385f8e47c2b.png">
